### ISSUE TITLE
Support `Invoke-WebRequest` over Windows named pipes

### ISSUE THREAD

**Issue by segevfiner** [PowerShell/PowerShell#24475]
### Summary of the new feature / enhancement

Docker on Windows listens on a named pipe for HTTP, and so do some other services. e.g. Node.js IPC servers. It would be nice to support this in `Invoke-WebRequest`.

P.S. This is also not supported in cURL, so I had to resort to using Node.js from the command line to send requests to such named pipe HTTP servers.

### Proposed technical implementation details (optional)

Basically the same as how `UnixSocket` is implemented, but using `NamedPipeClientStream` instead.

**Comment by SteveL-MSFT** [comment]
@PowerShell/wg-powershell-cmdlets discussed this.  Similar to adding UnixSockets support to the webcmdlets, we can see utility to adding named pipes support for Windows and believe the change is similar to the change for UnixSockets.

**Comment by sharktide** [comment]
@powershell/powershell Could I please be assigned?

**Comment by iSazonov** [comment]
@sharktide Feel free to grab. Please note that there is a risk that your PR will be frozen as it is not included in the current plans of MSFT team. The chances of acceptance will increase if you wrap it up as an experimental feature.

**Comment by sharktide** [comment]
@iSazonov How do I "wrap it up as an experimental feature". Also, do you know of any up-for-grabs issues that are in the current plans of the microsoft team?

**Comment by iSazonov** [comment]
Current MSFT team plan is in https://github.com/orgs/PowerShell/projects/44 I think there's nothing there for the community. Therefore, I warn you that the review may take a long time.
Examples for experimental features https://github.com/PowerShell/PowerShell/blob/master/test/powershell/engine/ExperimentalFeature/assets/ExpTest/ExpTest.cs
This is an isolation method for backward compatibility. It may not be necessary. So you can start without it.

**Comment by sharktide** [comment]
starting work...

**Comment by segevfiner** [comment]
Bump

**Comment by iSazonov** [comment]
There already is a PR for this.

**Comment by sharktide** [comment]
Bump

**Comment by segevfiner** [comment]
Bump
